test/metabase/query_processor/middleware/metrics_test.clj

```diff
@@ -692,3 +692,28 @@
                 (mt/format-rows-by
                  [u.date/temporal-str->iso8601-str int]
                  (mt/rows (qp/process-query query)))))))))
+
+(deftest ^:parallel metric-in-offset-test
+  (let [mp (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+        metric-query (-> (lib/query mp (lib.metadata/table mp (mt/id :orders)))
+                         (lib/aggregate (lib/count)))]
+    (mt/with-temp [:model/Card metric {:dataset_query (lib.convert/->legacy-MBQL metric-query)
+                                       :database_id (mt/id)
+                                       :name "Orders, Count"
+                                       :type :metric}]
+      (let [metric-meta (lib.metadata/metric mp (:id metric))
+            metric-offset #(lib/offset metric-meta -1)
```

Comment:
Why to define `metric-offset` as a function?

Reply:
Because we need to generate unique `:lib/uuid`s.